src/components/drawer/Drawer.tsx

```diff
@@ -233,19 +253,21 @@ const attachToDy = (animation: Animated.Value, newValue: number) => (
 const DrawerHeader = ({
   onClose,
   title,
+  titleIcon,
   isFullscreen,
   zIndex
 }: {
   onClose: () => void
   title?: string
+  titleIcon?: ImageSourcePropType
```

Comment:
there is a fun opportunity to strongly type the requirement that titleIcon can only be present when title is: https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase/#props-pass-one-only-if-the-other-is-passed